app/src/main/java/com/example/wellfed/ingredient/IngredientStorageFragment.java

```diff
@@ -19,184 +24,258 @@
+		}
+	});
+
+	/**
+	 * onCreate method for the IngredientStorageFragment.
+	 *
+	 * @param inflater           The LayoutInflater object that can be used
+	 *                           to inflate any views in the fragment.
+	 * @param container          If non-null, this is the parent view that
+	 *                           the fragment's UI should be attached to.
+	 * @param savedInstanceState If non-null, this fragment is being
+	 *                           re-constructed from a previous saved state
+	 *                           as given here.
+	 * @return Return the View for the fragment's UI, or null.
+	 */
+	@Nullable
+	@Override
+	public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+		foodStorage = new ArrayList<>();
+		controller = new IngredientStorageController(requireActivity());
+		controller.getAdapter().setOnItemClickListener(this);
+
+		return inflater.inflate(R.layout.fragment_ingredient_storage, container, false);
+	}
+
+	/**
+	 * onViewCreated method for the IngredientStorageFragment.
+	 *
+	 * @param view               The View returned by onCreateView.
+	 * @param savedInstanceState If non-null, this fragment is being
+	 *                           re-constructed from a previous saved state
+	 *                           as given here.
+	 */
+	@Override
+	public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+		super.onViewCreated(view, savedInstanceState);
+		recyclerView = view.findViewById(R.id.ingredient_storage_list);
+		recyclerView.setAdapter(controller.getAdapter());
+		recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
+
+		// Search bar
+		TextInputEditText searchBar = view.findViewById(R.id.ingredient_storage_search);
+		// Clear search bar
+		crossIcon = view.findViewById(R.id.clear_search);
+		crossIcon.setOnClickListener(v -> searchBar.setText(""));
+
+		// On search bar text change show "Functionality not implemented yet"
+		// message
+		searchBar.addTextChangedListener(new TextWatcher() {
+			@Override
+			public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+			}
+
+			@Override
+			public void onTextChanged(CharSequence s, int start, int before, int count) {
+				if (s.length() > 0) {
+					crossIcon.setVisibility(View.VISIBLE);
+				} else {
+					crossIcon.setVisibility(View.INVISIBLE);
+				}
+			}
+
+			@Override
+			public void afterTextChanged(Editable s) {
+				controller.getSearchResults(s.toString());
+			}
+		});
+
+		// Link filter button to filter functionality
+		imageFilterButton = view.findViewById(R.id.image_filter_button);
+		imageFilterButton.setOnClickListener(this::filter);
+	}
+
+	/**
+	 * Launches the IngredientEditActivity to edit an ingredient.
+	 *
+	 * @param pos The position of the ingredient in the list.
+	 */
+	@Override
+	public void launch(int pos) {
+		this.selected = pos;
+		launcher.launch(foodStorage.get(pos));
+	}
```

Comment:
We should switch this to using a specific StroageIngredient instead of its index.
TA said lists shouldn't be part of the UI classes.